aviatrix/resource_aviatrix_aws_tgw_vpn_conn.go

```diff
@@ -203,10 +220,52 @@ func resourceAviatrixAwsTgwVpnConnRead(d *schema.ResourceData, meta interface{})
 	d.Set("inside_ip_cidr_tun_1", vpnConn.InsideIpCIDRTun1)
 	d.Set("inside_ip_cidr_tun_2", vpnConn.InsideIpCIDRTun2)
 
+	if vpnConn.LearnedCidrsApproval == "yes" {
+		d.Set("enable_learned_cidrs_approval", true)
+	} else {
+		d.Set("enable_learned_cidrs_approval", false)
+	}
+
 	d.SetId(vpnConn.TgwName + "~" + vpnConn.VpnID)
 	return nil
 }
 
+func resourceAviatrixAwsTgwVpnConnUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	awsTgwVpnConn := &goaviatrix.AwsTgwVpnConn{
+		TgwName: d.Get("tgw_name").(string),
+		VpnID:   d.Get("vpn_id").(string),
+	}
+
+	d.Partial(true)
+	log.Printf("[INFO] Updating Aviatrix aws tgw vpn connection: %#v", awsTgwVpnConn)
+
+	if d.HasChange("enable_learned_cidrs_approval") {
+		if d.Get("connection_type").(string) == "static" {
+			return fmt.Errorf("learned cidrs approval is supported for a BGP VPN connection, not for a static connection")
+		}
+		learnedCidrsApproval := d.Get("enable_learned_cidrs_approval").(bool)
+		if learnedCidrsApproval {
+			awsTgwVpnConn.LearnedCidrsApproval = "yes"
+			err := client.EnableVpnConnectionLearnedCidrsApproval(awsTgwVpnConn)
+			if err != nil {
+				return fmt.Errorf("failed to enable learned cidrs approval: %s", err)
+			}
+		} else {
+			awsTgwVpnConn.LearnedCidrsApproval = "no"
+			err := client.DisableVpnConnectionLearnedCidrsApproval(awsTgwVpnConn)
+			if err != nil {
+				return fmt.Errorf("failed to disable learned cidrs approval: %s", err)
+			}
+		}
+		d.SetPartial("enable_learned_cidrs_approval")
+	}
+
+	d.SetId(awsTgwVpnConn.TgwName + "~" + awsTgwVpnConn.VpnID)
+	return resourceAviatrixAwsTgwVpnConnRead(d, meta)
```

Comment:
use the defered read to handle failures

Reply:
Only one thing, which is to create an aws_tgw_vpn_conn, is done in create function. Deferred read is not needed for this case.

Reply:
What if the disable actions fails but gives a warning and the state is actually false?

You can avoid the additional terraform refresh

Reply:
It's a good practice to follow everywhere